internal/cluster/client_message_test.go

```diff
@@ -0,0 +1,150 @@
+package cluster
+
+import (
+	"bytes"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"testing"
+)
+
+func TestClientMessageReader(t *testing.T) {
+	testCases := []struct {
+		f    func(t *testing.T)
+		name string
+	}{
+		{name: "readSingleFrame", f: readSingleFrame},
+		{name: "readMultiFrameMessage", f: readMultiFrameMessage},
+		{name: "readFramesInMultipleCallsToRead", f: readFramesInMultipleCallsToRead},
+		{name: "readWhenTheFrameLengthAndFlagsNotReceivedAtFirst", f: readWhenTheFrameLengthAndFlagsNotReceivedAtFirst},
+		{name: "readFramesInMultipleCallsToReadFromWhenLastPieceIsSmall", f: readFramesInMultipleCallsToReadFromWhenLastPieceIsSmall},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.f(t)
+		})
+	}
+}
+
+func readSingleFrame(t *testing.T) {
```

Comment:
Could you add the `Test` suffix to test functions?

Reply:
Could you rename this test to `readSingleFrameMessageTest` to make it more similar to the reference implementation counterpart ?

Reply:
Added at https://github.com/hazelcast/hazelcast-go-client/pull/863/commits/c2d479a6e1ae3cc63c216397d2170510eb315fb6.